apps/emqx_bridge_tdengine/src/emqx_bridge_tdengine_connector.erl

```diff
@@ -203,6 +206,37 @@ do_query(InstanceId, Query, #{pool_name := PoolName, query_opts := Opts} = State
 execute(Conn, Query, Opts) ->
     tdengine:insert(Conn, Query, Opts).
 
+do_batch_insert(Conn, Tokens, BatchReqs, Opts) ->
+    Queries = aggregate_query(Tokens, BatchReqs),
+    SQL = lists:foldl(
+        fun({InsertPart, Values}, Acc) ->
+            lists:foldl(
+                fun(ValuePart, IAcc) ->
+                    <<IAcc/binary, " ", ValuePart/binary>>
+                end,
+                <<Acc/binary, " ", InsertPart/binary, " VALUES">>,
+                Values
+            )
+        end,
+        <<"INSERT INTO">>,
+        Queries
+    ),
+    execute(Conn, SQL, Opts).
+
+aggregate_query({InsertPartTks, ParamsPartTks}, BatchReqs) ->
+    maps:to_list(
+        lists:foldl(
```

Comment:
Q: would it make sense in this case for the fold to accumulate directly into a list rather than going through a map and then back?

Reply:
No, this function aggregates the queries by the table name, at this point, the list is slower than the map when lookup or updating by a key.
But the `maps:to_list` isn't necessary, I have removed it.